src/main/scala/com/microsoft/ml/spark/cognitive/TextAnalyticsSDK.scala

```diff
@@ -18,20 +17,23 @@ import org.apache.spark.sql.functions.col
 import org.apache.spark.sql.types.{DataTypes, StructType}
 import org.apache.spark.sql.{DataFrame, Dataset, Row}
 import com.azure.ai.textanalytics.models
+import org.apache.spark.sql.Encoder
 
 import java.net.URI
 import scala.collection.JavaConverters._
 
 abstract class TextAnalyticsSDKBase[T](val textAnalyticsOptions: Option[TextAnalyticsRequestOptions] = None)
   extends Transformer
   with HasInputCol with HasErrorCol
-  with HasEndpoint with HasSubscriptionKey
+  with HasEndpoint with HasSubscriptionKey with HasOutputCol
   with ComplexParamsWritable with BasicLogging {
 
   protected val invokeTextAnalytics: String => TAResponseV4[T]
 
   protected def outputSchema: StructType

```

Comment:
Note that this defines an abstract function, which means every time you reference `outputSchema` it will invoke a method which will return the StructType. This can have performance impact. 
It's better to have this as a value `val` instead, so there's only one schema copy for every invocation.

Reply:
When trying to change to val, I have many errors. Will working on implementing again for new PR.